File: Client/src/main/java/com/rspsi/editor/model/WorldModel.java
package com.rspsi.editor.model;

/**
 * @deprecated use {@link WorldDocument}. Kept as a source-compatible bridge
 * while existing callers move to the canonical document name.
 */
@Deprecated
public final class WorldModel extends WorldDocument {
    public static final int DEFAULT_PLANES = WorldDocument.DEFAULT_PLANES;

    public WorldModel(int width, int length) {
        super(width, length);
    }

    public WorldModel(int width, int length, int planes) {
        super(width, length, planes);
    }
}

File: Client/src/main/kotlin/com/rspsi/editor/model/WorldModel.kt
package com.rspsi.editor.model

/**
 * Deprecated source-compatible bridge to the canonical [WorldDocument] name.
 *
 * New code should construct [WorldDocument] directly. This type remains only while older callers
 * complete that rename, so its constructors, inheritance, static default-plane constant, and
 * deprecation metadata are intentionally preserved.
 */
@kotlin.Deprecated("Use WorldDocument")
@java.lang.Deprecated
class WorldModel : WorldDocument {
    constructor(
        width: Int,
        length: Int,
    ) : super(width, length)

    constructor(
        width: Int,
        length: Int,
        planes: Int,
    ) : super(width, length, planes)

    companion object {
        const val DEFAULT_PLANES: Int = WorldDocument.DEFAULT_PLANES
    }
}
